### ISSUE TITLE
IMAP error on tidy

### ISSUE THREAD

**Issue by ppawlak** [sapristi/mail_juicer#3]
Hello,

When I run the tidy command I have the following error.
Any idea?

```
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮                                                                                      [93/1032]
│ /home/patrice/.pyenv/versions/3.12.0/envs/mmuxer/lib/python3.12/site-packages/mmuxer/cli/run.py: │
│ 26 in tidy                                                                                       │
│                                                                                                  │
│   23 │   if folder is not None:                                                                  │
│   24 │   │   box.folder.set(folder)                                                              │
│   25 │   counter = 0                                                                             │
│ ❱ 26 │   for msg in box.fetch(bulk=True):                                                        │
│   27 │   │   msg.associated_folder = folder                                                      │
│   28 │   │   apply_list(state.rules, box, msg, dry_run)                                          │
│   29 │   │   for script in state.scripts:                                                        │
│                                                                                                  │
│ ╭────────────────────────────── locals ───────────────────────────────╮                          │
│ │         box = <imap_tools.mailbox.MailBox object at 0x7ff1a4507170> │                          │
│ │ config_file = PosixPath('/home/patrice/mmuxer/config.yaml')         │                          │
│ │     counter = 0                                                     │                          │
│ │     dry_run = False                                                 │                          │
│ │      folder = None                                                  │                          │
│ ╰─────────────────────────────────────────────────────────────────────╯                          │
│                                                                                                  │
│ /home/patrice/.pyenv/versions/3.12.0/envs/mmuxer/lib/python3.12/site-packages/imap_tools/mailbox │
│ .py:161 in fetch                                                                                 │
│                                                                                                  │
│   158 │   │   limit_range = slice(0, limit) if type(limit) is int else limit or slice(None)      │
│   159 │   │   assert type(limit_range) is slice                                                  │
│   160 │   │   uids = tuple((reversed if reverse else iter)(self.uids(criteria, charset)))[limi   │
│ ❱ 161 │   │   for fetch_item in (self._fetch_in_bulk if bulk else self._fetch_by_one)(uids, me   │
│   162 │   │   │   yield self.email_message_class(fetch_item)                                     │
│   163 │                                                                                          │
│   164 │   def expunge(self) -> tuple:                                                            │
│                                                                                                  │
│ ╭─────────────────────────────── locals ────────────────────────────────╮                        │
│ │          bulk = True                                                  │                        │
│ │       charset = 'US-ASCII'                                            │                        │
│ │      criteria = 'ALL'                                                 │                        │
│ │  headers_only = False                                                 │                        │
│ │         limit = None                                                  │                        │
│ │   limit_range = slice(None, None, None)                               │                        │
│ │     mark_seen = True                                                  │                        │
│ │ message_parts = '(BODY[] UID FLAGS RFC822.SIZE)'                      │                        │
│ │       reverse = False                                                 │                        │
│ │          self = <imap_tools.mailbox.MailBox object at 0x7ff1a4507170> │                        │
│ │          uids = (                                                     │                        │
│ │                 │   '277',                                            │                        │
│ │                 │   '278',                                            │                        │
│ │                 │   '282',                                            │                        │
│ │                 │   '283',                                            │                        │
│ │                 │   '284',                                            │                        │
│ │                 │   '289',                                            │                        │
│ │                 │   '290',                                            │                        │
│ │                 │   '291',                                            │                        │
│ │                 │   '292',                                            │                        │
│ │                 │   '293',                                            │                        │
│ │                 │   ... +11440                                        │                        │
│ │                 )                                                     │                        │
│ ╰───────────────────────────────────────────────────────────────────────╯                        │
│                                                                                                  │

/home/patrice/.pyenv/versions/3.12.0/envs/mmuxer/lib/python3.12/site-packages/imap_tools/mailbox │
│ .py:134 in _fetch_in_bulk                                                                        │
│                                                                                                  │
│   131 │   def _fetch_in_bulk(self, uid_list: Sequence[str], message_parts: str, reverse: bool)   │
│   132 │   │   if not uid_list:                                                                   │
│   133 │   │   │   return                                                                         │
│ ❱ 134 │   │   fetch_result = self.client.uid('fetch', ','.join(uid_list), message_parts)         │
│   135 │   │   check_command_status(fetch_result, MailboxFetchError)                              │
│   136 │   │   if not fetch_result[1] or fetch_result[1][0] is None:                              │
│   137 │   │   │   return                                                                         │
│                                                                                                  │
│ ╭─────────────────────────────── locals ────────────────────────────────╮                        │
│ │ message_parts = '(BODY[] UID FLAGS RFC822.SIZE)'                      │                        │
│ │       reverse = False                                                 │                        │
│ │          self = <imap_tools.mailbox.MailBox object at 0x7ff1a4507170> │                        │
│ │      uid_list = (                                                     │                        │
│ │                 │   '277',                                            │                        │
│ │                 │   '278',                                            │                        │
│ │                 │   '282',                                            │                        │
│ │                 │   '283',                                            │                        │
│ │                 │   '284',                                            │                        │
│ │                 │   '289',                                            │                        │
│ │                 │   '290',                                            │                        │
│ │                 │   '291',                                            │                        │
│ │                 │   '292',                                            │                        │
│ │                 │   '293',                                            │                        │
│ │                 │   ... +11440                                        │                        │
│ │                 )                                                     │                        │
│ ╰───────────────────────────────────────────────────────────────────────╯                        │
│                                                                                                  │
│ /home/patrice/.pyenv/versions/3.12.0/lib/python3.12/imaplib.py:890 in uid                        │
│                                                                                                  │
│    887 │   │   │   │   │   │   │    (command, self.state,                                        │
│    888 │   │   │   │   │   │   │     ', '.join(Commands[command])))                              │
│    889 │   │   name = 'UID'                                                                      │
│ ❱  890 │   │   typ, dat = self._simple_command(name, command, *args)                             │
│    891 │   │   if command in ('SEARCH', 'SORT', 'THREAD'):                                       │
│    892 │   │   │   name = command                                                                │
│    893 │   │   else:                                                                             │

│ ╭─────────────────────────────────────────── locals ───────────────────────────────────────────╮ │
│ │    args = (                                                                                  │ │
│ │           │                                                                                  │ │
│ │           '277,278,282,283,284,289,290,291,292,293,296,297,300,301,302,303,304,307,308,309,… │ │
│ │           │   '(BODY[] UID FLAGS RFC822.SIZE)'                                               │ │
│ │           )                                                                                  │ │
│ │ command = 'FETCH'                                                                            │ │
│ │    name = 'UID'                                                                              │ │
│ │    self = <imaplib.IMAP4_SSL object at 0x7ff1a3045d90>                                       │ │
│ ╰──────────────────────────────────────────────────────────────────────────────────────────────╯ │
│                                                                                                  │
│ /home/patrice/.pyenv/versions/3.12.0/lib/python3.12/imaplib.py:1230 in _simple_command           │
│                                                                                                  │
│   1227 │                                                                                         │
│   1228 │   def _simple_command(self, name, *args):                                               │
│   1229 │   │                                                                                     │
│ ❱ 1230 │   │   return self._command_complete(name, self._command(name, *args))                   │
│   1231 │                                                                                         │
│   1232 │                                                                                         │
│   1233 │   def _untagged_response(self, typ, dat, name):                                         │
│                                                                                                  │
│ ╭─────────────────────────────────────────── locals ───────────────────────────────────────────╮ │
│ │ args = (                                                                                     │ │
│ │        │   'FETCH',                                                                          │ │
│ │        │                                                                                     │ │
│ │        '277,278,282,283,284,289,290,291,292,293,296,297,300,301,302,303,304,307,308,309,'+6… │ │
│ │        │   '(BODY[] UID FLAGS RFC822.SIZE)'                                                  │ │
│ │        )                                                                                     │ │
│ │ name = 'UID'                                                                                 │ │
│ │ self = <imaplib.IMAP4_SSL object at 0x7ff1a3045d90>                                          │ │
│ ╰──────────────────────────────────────────────────────────────────────────────────────────────╯ │
│                                                                                                  │
│ /home/patrice/.pyenv/versions/3.12.0/lib/python3.12/imaplib.py:1055 in _command_complete         │
│                                                                                                  │
│   1052 │   │   if not logout:                                                                    │
│   1053 │   │   │   self._check_bye()                                                             │
│   1054 │   │   if typ == 'BAD':                                                                  │
│ ❱ 1055 │   │   │   raise self.error('%s command error: %s %s' % (name, typ, data))               │
│   1056 │   │   return typ, data                                                                  │
│   1057                                                                                           │
│   1058                                                                                           │
│                                                                                                  │
│ ╭───────────────────────── locals ─────────────────────────╮                                     │
│ │   data = [b'parse error: maximum request size exceeded'] │                                     │
│ │ logout = False                                           │                                     │
│ │   name = 'UID'                                           │                                     │
│ │   self = <imaplib.IMAP4_SSL object at 0x7ff1a3045d90>    │                                     │
│ │    tag = b'ODOL4'                                        │                                     │
│ │    typ = 'BAD'                                           │                                     │
│ ╰──────────────────────────────────────────────────────────╯                                     │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
error: UID command error: BAD [b'parse error: maximum request size exceeded']
```

**Comment by sapristi** [comment]
Hello, this seem related to fetching emails in bulk

I have published the version `0.6.0`,  with an additional setting to limit batch size. It's set to 100, which I think should be ok. If you still have the error, you can set it in the config file, for example

```yml
settings:
  server: ---
  username: ---
  fetch_batch_size: 10
```

**Comment by ppawlak** [comment]
Thanks for the quick reply and update @sapristi !

Version 0.6.0 does fix the error indeed but it seems to only deal with a 100 messages and not iterate over the rest of the messages in my inbox:

```
$ mmuxer tidy --config-file config.yaml
INFO     Loading config from /home/patrice/mmuxer/config.yaml
INFO     Connected to xxx with xxx

100 messages parsed.
```

Or am I missing something?

**Comment by sapristi** [comment]
Hmnm indeed, that was not expected, I'll see what  i can do! (I have removed v0.6.0 from pypi)

**Comment by sapristi** [comment]
Ok published version  `0.6.0a3`, this should be ok this time :)

**Comment by ppawlak** [comment]
I do not see version `0.6.0a3` on PyPI  (last available is `0.5.3` at this time) but I installed from Github and it does work.

Thanks a lot!

**Comment by sapristi** [comment]
Ok thanks for the update :)  here's the version, just in case https://pypi.org/project/mmuxer/0.6.0a3/
